MSA-2084: Spruce Sensor DTH new model and health check

natec007 opened the pull request:
Submitting 'Spruce Sensor DTH new model and health check' for publication

natec007 committed:
MSA-2084: Updated with new model number and health check

jackchi commented on devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy:
```diff
@@ -14,25 +14,34 @@
+        capability "Health Check"
         //capability "Polling"
 		
+        attribute "checkInterval", "string"
```
No need to add `checkInterval` here. It's actually included as a `NUMBER` in the Capbility `Health Check`

jackchi commented on devicetypes/plaidsystems/spruce-sensor.src/spruce-sensor.groovy:
```diff
@@ -293,6 +302,11 @@ def setConfig(){
 
+def installed(){
+	//check every 1 hour + 2mins
+    sendEvent(name: "checkInterval", value: 1 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
```
Let's put this `sendEvent("checkInterval")` in both `installed()` & `updated()`

natec007 replied:
Its a sleepy ZigBee device, so can't be pinged except when awake.
Made the changes requested

jackchi replied:
So how often does this device check-in (generate an event) when idle? Currently if our platform doesn't hear from this device in 62min (your checkInterval value) it'll mark it `OFFLINE`

natec007 replied:
It pings the coordinator (hub) every 10 mins but reports data between 10-120mins.  So if the ZigBee event is caught then 62mins should be fine, if you only look at reported data, it should change to 2 hours.

jackchi replied:
👍

jackchi commented:
Is this DTH "pingable()" - meaning, that we can contact the device and device would generate an event in the parse() method?

jackchi reviewed (changes requested):
1. Take out `checkInterval` from custom attribute
2. Add `sendEvent("checkInterval")` to both updated/installed
3. Can this device be pinged?

natec007 committed:
Update spruce-sensor.groovy

Removed checkInterval attribute
sendEvent(name: "checkInterval" already in updated and installed

jackchi reviewed (approved):
Everything looks good.
